src/main/java/uk/gov/ons/ctp/common/event/EventBuilder.java

```diff
@@ -0,0 +1,404 @@
+    SendInfo create(String json) {
+      GenericEvent genericEvent = deserialiseEventJson(json, RespondentAuthenticatedEvent.class);
+      EventPayload payload =
+          ((RespondentAuthenticatedEvent) genericEvent).getPayload().getResponse();
+      return build(genericEvent, payload);
+    }
+  }
+
+  public abstract static class CaseBuilder extends EventBuilder {
+    @Override
+    GenericEvent create(SendInfo sendInfo) {
+      CaseEvent caseEvent = new CaseEvent();
+      caseEvent.setEvent(buildHeader(type(), sendInfo.getSource(), sendInfo.getChannel()));
+      CasePayload casePayload = new CasePayload((CollectionCase) sendInfo.getPayload());
+      caseEvent.setPayload(casePayload);
+      return caseEvent;
+    }
+
+    @Override
+    SendInfo create(String json) {
+      GenericEvent genericEvent = deserialiseEventJson(json, CaseEvent.class);
+      EventPayload payload = ((CaseEvent) genericEvent).getPayload().getCollectionCase();
+      return build(genericEvent, payload);
+    }
+
+    abstract EventType type();
+  }
+
+  public static class CaseCreatedBuilder extends CaseBuilder {
+    @Override
+    EventType type() {
+      return EventType.CASE_CREATED;
+    }
+  }
+
+  public static class CaseUpdatedBuilder extends CaseBuilder {
+    @Override
+    EventType type() {
+      return EventType.CASE_UPDATED;
+    }
+  }
+
+  public static class RefusalReceivedBuilder extends EventBuilder {
+    @Override
+    GenericEvent create(SendInfo sendInfo) {
+      RespondentRefusalEvent respondentRefusalEvent = new RespondentRefusalEvent();
+      respondentRefusalEvent.setEvent(
+          buildHeader(EventType.REFUSAL_RECEIVED, sendInfo.getSource(), sendInfo.getChannel()));
+      RespondentRefusalPayload respondentRefusalPayload =
+          new RespondentRefusalPayload((RespondentRefusalDetails) sendInfo.getPayload());
+      respondentRefusalEvent.setPayload(respondentRefusalPayload);
+      return respondentRefusalEvent;
+    }
+
+    @Override
+    SendInfo create(String json) {
+      GenericEvent genericEvent = deserialiseEventJson(json, RespondentRefusalEvent.class);
+      EventPayload payload = ((RespondentRefusalEvent) genericEvent).getPayload().getRefusal();
+      return build(genericEvent, payload);
+    }
+  }
+
+  public abstract static class UacBuilder extends EventBuilder {
+    @Override
+    GenericEvent create(SendInfo sendInfo) {
+      UACEvent uacEvent = new UACEvent();
+      uacEvent.setEvent(buildHeader(type(), sendInfo.getSource(), sendInfo.getChannel()));
+      UACPayload uacPayload = new UACPayload((UAC) sendInfo.getPayload());
+      uacEvent.setPayload(uacPayload);
+      return uacEvent;
+    }
+
+    @Override
+    SendInfo create(String json) {
+      GenericEvent genericEvent = deserialiseEventJson(json, UACEvent.class);
+      EventPayload payload = ((UACEvent) genericEvent).getPayload().getUac();
+      return build(genericEvent, payload);
+    }
+
+    abstract EventType type();
```

Comment:
I can see that UacBuilder and CaseBuilder reduce a bit of code duplication but it does create a break in the flow of small and simple classes which all look alike. 
The overall line saving is very modest. I'd have thought that the 'dumb' version would be simpler for the reader to get their head round.

Reply:
Change made as suggested. Now it is less clever, but perhaps more readable.